src/ImageProcessorCore.Web/Middleware/ImageProcessorMiddleware.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.AspNet.Http;
+using System.IO;
+using ImageProcessorCore.Samplers;
+
+namespace ImageProcessorCore.Web.Middleware
+{
+    public class ImageProcessorMiddleware
+    {
+        private readonly IApplicationEnvironment _appEnvironment;
+
+        private readonly RequestDelegate _next;
+
+        public ImageProcessorMiddleware(RequestDelegate next, IApplicationEnvironment appEnvironment)
+        {
+            _next = next;
+            _appEnvironment = appEnvironment;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (!IsImageRequest(context))
+            {
+                // move to the next request here?
+                await _next.Invoke(context);
+                return;
+            }
+
+            try
+            {
+                //var sb = new StringBuilder();
+                //sb.AppendLine("=====================================================================================================");
+                //sb.AppendLine("Date/Time: " + DateTime.UtcNow.ToString("M/d/yyyy hh:mm tt"));
+
+                int width = 0;
+                int.TryParse(context.Request.Query["width"], out width);
+                int height = 0;
+                int.TryParse(context.Request.Query["height"], out height);
+
+                var inputPath = _appEnvironment.ApplicationBasePath + "/wwwroot" + context.Request.Path;
+                using (var inputStream = File.OpenRead(inputPath))
+                {
+                    //sb.AppendLine("Input Path: " + inputPath);
+                    //sb.AppendLine("Querystring Dimensions: " + width.ToString() + "x" + height.ToString());
+
+                    //var sw = new Stopwatch();
+                    //sw.Start();
+                    var image = new Image(inputStream);
+                    if (image.Width > image.Height && width != 0)
+                        height = (width * image.Height) / image.Width;
+                    if (image.Height > image.Width && height != 0)
+                        width = (height * image.Width) / image.Height;
+                    //sw.Stop();
+
+                    //sb.AppendLine("Original Dimensions: " + image.Width.ToString() + "x" + image.Height.ToString());
+                    //sb.AppendLine("Output Dimensions: " + width.ToString() + "x" + height.ToString());
+                    //sb.AppendLine("Image Read Time in Seconds: " + sw.Elapsed.TotalSeconds.ToString());
+
+                    // write directly to the body output stream
+                    using (var outputStream = new MemoryStream())
+                    {
+                        //sw.Restart();
+                        image.Resize(width, height)
```

Comment:
This will all need to be abstracted out to use the extension method 

    Process(this Image source, params IImageProcessor[] processors)